scripts/utils/docs-content-tools/create-nav-yaml.js

```diff
@@ -0,0 +1,159 @@
+const fs = require('fs');
+const dirTree = require('directory-tree');
+const yaml = require('js-yaml');
+
+const WRITE_DIRECTORY = './src/docs-content-nav/';
+const READ_DIRECTORY = './src/content/docs/';
+
+/**
+ * Strip the file extension from the path of the file
+ * @param {string} path The path to strip the extension from.
+ * @returns {string} The path without the file extension.
```

Comment:
Suuuuper nitpicky: can we make these capital case `String` types? We seem to be doing that in the other type JSDoc comments.

Reply:
Yes that is a well picked nit